staging/src/k8s.io/apimachinery/pkg/util/jsonmergepatch/patch_test.go

```diff
@@ -88,19 +88,23 @@ testCases:
       mapKey:
         nilKey1: null
         nilKey2: null
-    expectedWithoutNull:
-      mapKey: {}
+    expectedWithoutNull: {}
   - description: nested map that all subkeys are non-nil
     originalObj:
       mapKey:
         nonNilKey: foo
         nonNilKey: bar
-    expectedWithNull:
-      mapKey: {}
+    expectedWithNull: {}
     expectedWithoutNull:
       mapKey:
         nonNilKey: foo
         nonNilKey: bar
+  - description: nested map that is explicitly empty
```

Comment:
This is not a nested map, please fix the description.
And we should have a few more test cases:
```
-   originalObj:
      mapKey:
           nonNilKey: {}
-   originalObj:
      mapKey:
           nonNilKey1: {}
           nonNilKey2: {}
-   originalObj:
      mapKey:
           nonNilKey: {}
           nilKey: null
```

Reply:
technically it's an empty map that's nested in another map, but I'll change to make it clearer

I'll add the suggested testcases